worker-message-prioritization-rerouting/src/main/java/com/github/workerframework/workermessageprioritization/rerouting/StagingQueueCreator.java

```diff
@@ -101,7 +111,15 @@ void createStagingQueue(final Queue targetQueue, final String stagingQueueName)
             // Throw original exception
             throw ioException;
         }
+    }
 
-        declaredQueues.add(stagingQueueName);
+    private List<String> getPossiblyCachedStagingQueueNames()
+    {
+        return cachingQueuesApi.getApi()
```

Comment:
Need to verify the performance of this, will every call to getPossiblyCachedStagingQueueNames result in the cached JSON string being deserialised into Java objects?
The concern I mentioned, when we spoke last week, is that some worker process messages at 100/s and this deserialisation will be expensive.

Reply:
Yes the response is deserialized into Java objects, even when the response is served from the cache.

Should we just [go back to using the LoadingCache](https://github.com/WorkerFramework/worker-message-prioritization/pull/20/commits/a65785f4d21d2e140eee4fce5bc90471577f17ba) which doesn't have this problem?

Reply:
If there isn't a way to have retrofit cache the deserialized list then perhaps try
https://www.baeldung.com/guava-memoizer